lib/patchestry/intrinsics/patch_utils.c

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (c) 2025, Trail of Bits, Inc.
+ *
+ * This source code is licensed in accordance with the terms specified in
+ * the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <patchestry/intrinsics/patchestry_intrinsics.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <pthread.h>
+#include <dlfcn.h>
+#include <unistd.h>
+
+// =============================================================================
+// Patch Management Implementation
+// =============================================================================
+
+typedef struct patch_entry {
+    char name[64];
+    char version[32];
+    char target_function[64];
+    patchestry_patch_state_t state;
+    void* patch_data;
+    size_t data_size;
+    
+    // Dependencies and conflicts
+    char dependencies[8][64];
+    int dependency_count;
+    char conflicts[8][64];
+    int conflict_count;
+    
+    struct patch_entry* next;
+} patch_entry_t;
+
+static patch_entry_t* patch_list = NULL;
+static pthread_mutex_t patch_mutex = PTHREAD_MUTEX_INITIALIZER;
+
+static patch_entry_t* find_patch_entry(const char* patch_name) {
+    patch_entry_t* entry = patch_list;
+    while (entry) {
+        if (strcmp(entry->name, patch_name) == 0) {
```

Comment:
Should this be `strncmp` or another string comparator function which is a bit better guarded? I believe `strcmp` runs until it sees a null terminator in either string, which is a bit scary as I don't think as Patchestry we technically control the entire life cycle of either of the strings being compared here